gsadjust/gui/tabs/drift.py

```diff
@@ -246,9 +259,17 @@ def reset(self):
         self.axes_drift_cont_upper.figure.canvas.draw()
         self.axes_drift_cont_lower.figure.canvas.draw()
         self.axes_drift_single.figure.canvas.draw()
-        self.delta_view.setModel(DeltaTableModel())

```

Comment:
These lines changed the model on the view, but do not change what is pointed to e.g. by `self.dg_samples_table` -- so you now have two models existing at the same time.  This might have been throwing the errors due to mismatches between items.

The `DeltaTableModel()` call doesn't keep a reference to that model, I'm 50:50 on whether that means the Python part would be garbage collected. If that happened you'd see segfaults.